packages/EasyCore/src/Doctrine/Dispatchers/DeferredEntityEventDispatcher.php

```diff
@@ -93,36 +112,73 @@ public function disable(): void
 
     public function dispatch(): void
     {
-        $entityInsertions = $this->entityInsertions;
-        $entityUpdates = $this->entityUpdates;
-
-        $this->clear();
-
         if ($this->enabled === false) {
             return;
         }
 
-        $processedEntities = [];
-        foreach ($entityInsertions as $entities) {
-            foreach ($entities as $oid => $entity) {
-                $processedEntities[$oid] = $entity;
-                $this->eventDispatcher->dispatch(new EntityCreatedEvent($entity));
+        try {
+            $mergedEntityChangeSets = [];
+            foreach ($this->entityChangeSets as $levelChangeSets) {
+                foreach ($levelChangeSets as $oid => $changeSet) {
+                    $mergedEntityChangeSets[$oid] = $this->mergeChangeSet(
+                        $mergedEntityChangeSets[$oid] ?? [],
+                        $changeSet
+                    );
+                }
             }
-        }
 
-        foreach ($entityUpdates as $entities) {
-            foreach ($entities as $oid => $entity) {
-                if (isset($processedEntities[$oid])) {
-                    continue;
-                }
-                $processedEntities[$oid] = $entity;
-                $this->eventDispatcher->dispatch(new EntityUpdatedEvent($entity));
+            foreach ($mergedEntityChangeSets as $oid => $entityChangeSet) {
+                $event = $this->createEntityEvent($oid, $entityChangeSet);
+
+                $this->eventDispatcher->dispatch($event);
             }
+        } finally {
+            $this->clear();
         }
     }
 
     public function enable(): void
     {
         $this->enabled = true;
     }
+
+    /**
+     * @param string $oid
+     * @param array<string, array{mixed, mixed}> $entityChangeSet
+     *
+     * @return \EonX\EasyCore\Doctrine\Events\EntityCreatedEvent|\EonX\EasyCore\Doctrine\Events\EntityUpdatedEvent
+     */
+    private function createEntityEvent(string $oid, array $entityChangeSet)
+    {
+        if (isset($this->entityInsertions[$oid]) !== false) {
+            return new EntityCreatedEvent($this->entityInsertions[$oid], $entityChangeSet);
+        }
+
+        if (isset($this->entityUpdates[$oid]) !== false) {
+            return new EntityUpdatedEvent($this->entityUpdates[$oid], $entityChangeSet);
+        }
+
+        // @codeCoverageIgnoreStart
+        throw new \LogicException('Entity for event not found.');
```

Comment:
This class needs to be imported.

Reply:
Fixed in `EasyDoctrine` package - https://github.com/eonx-com/easy-monorepo/blob/33c75a99d89874fa21b7e1e78c8fca85c6a37332/packages/EasyDoctrine/src/Dispatchers/DeferredEntityEventDispatcher.php#L9